doc/users/next_whats_new/striped_lines.rst

```diff
@@ -0,0 +1,19 @@
+Striped lines (experimental)
+----------------------------
+
+New *second_color* parameter enables the creation of striped lines.
+
+.. plot::
+    :include-source: true
+
+    import matplotlib.pyplot as plt
+    import numpy as np
+
+    x = np.linspace(1., 3., 10)
+    y = x**3
+
+    fig, ax = plt.subplots()
+    ax.plot(x, y, linestyle='--', color='orange', second_color='blue',
```

Comment:
**TL;DR I think we should go with `gapcolor`.**

**I'm very sorry for going back and forth, but I just realized how this parameter fits into the whole color story.**


Seeing this makes me wonder whether `second_color` is actually a good name Because for 
```
    ax.plot(x, y, linestyle='o', color='orange', second_color='blue',
```
`color` affects the marker but `second_color` does not. This is a certain asymmetry.

Actually the color mechanism is somewhat complicated: 

| **Visual part of Line2D artist**                 | **explicit parameter** | **Affected by `color`** |
|--------------------------------------------------|------------------------|-------------------------|
| marker face                                      | markerfacecolor        | x                       |
| additional marker face (for half-filled markers) | markerfacecoloralt     | -                       |
| marker edge                                      | markeredgecolor        | x                       |
| line                                             | -                      | x                       |
| this PR: additional line part (for dashed lines) | second_color           | -                       |

Observations:

- `color` is a global parameter that can affect multiple visual parts. The behavior is reaonable: Color all visible parts with this color". (It's correct that this does not apply to the additional marker face of half-filled markers, because that would make them fully filled.)
- Conceptually the `gap_color` (I call it like this for now) is similar to `markerfacecoloralt` - it should and isn't affected by `color`.
- We don't have an explicit `linecolor` parameter. That's also ok. Think: We start with a base `color` and one can optionally change all marker parameters on top of this.

When only thinking about the line part `color` + `second_color` is a perfectly good API. However, taking the whole picture into account, it's not specific enough. IMHO the name has to contain the visual part that is addressed. My suggestions in order of preference:

- `gapcolor`
- `secondary_line_color` (very clear, but very long).
- `linecoloralt` (rhymes with markerfacecoloralt, but a bit awkward to have `...alt` without a `linecolor` and I don't think we need to add that).

Reply:
Way too late to change I know, but I don't think it would be bad to have a linecolor alias  b/c in the marker+line case, is kinda confusing that the line color is color but the markercolor is markerfacecolor + markeredgecolor (also that there's no markercolor)

Reply:
I think if we had `second_line_color` then a user might expect there to be a `line_color`.  Someone (@efiring I think) mentioned that “alt color” has problematic meaning in some places.

So `gapcolor` also seems best to me.